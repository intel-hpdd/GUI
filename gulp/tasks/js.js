//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
