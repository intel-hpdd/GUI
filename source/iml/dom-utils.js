// @flow

type QuerySelector<T: HTMLElement> = (HTMLElement | Document, string) => T;

export const querySelector: QuerySelector<*> = (
  elm: HTMLElement | Document,
  sel: string
) => {
  const selected = elm.querySelector(sel);

  if (!selected)
    throw new Error(
      `Could not find selector ${sel} for element ${elm.nodeName}`
    );

  return selected;
};
