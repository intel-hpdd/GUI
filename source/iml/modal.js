// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import Inferno from 'inferno';

type headerT = {
  children?: React$Element<*>
};

export const Header = ({ children }: headerT) => {
  return <div class="modal-header">{children}</div>;
};

type bodyT = {
  moreClasses?: string[],
  children?: React$Element<*>
};

export const Body = ({ children, moreClasses = [] }: bodyT) => (
  <div className={`modal-body ${moreClasses.join(' ')}`}>{children}</div>
);

type footerT = {
  children?: React$Element<*>
};

export const Footer = ({ children }: footerT) => {
  return <div class="modal-footer">{children}</div>;
};

type modalT = {
  children?: React$Element<*>,
  moreClasses?: string[],
  visible: boolean
};

export const Modal = ({ moreClasses = [], children, visible }: modalT) => {
  if (!visible) return;

  return (
    <div style={{ display: 'block' }} tabindex="-1" role="dialog" className={`modal fade in ${moreClasses.join(' ')}`}>
      <div class="modal-dialog" role="document">
        <div class="modal-content">{children}</div>
      </div>
    </div>
  );
};
