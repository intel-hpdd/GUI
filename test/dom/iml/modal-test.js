// @flow

import { render } from "inferno";
import { Modal, Header, Body, Footer, Backdrop } from "../../../source/iml/modal.js";
import { querySelector } from "../../../source/iml/dom-utils.js";

describe("Modal test", () => {
  let root: HTMLDivElement, onAgree, onDisagree, TestModal, el: HTMLElement;
  beforeEach(() => {
    root = document.createElement("div");
    querySelector(document, "body").appendChild(root);
    onAgree = jest.fn();
    onDisagree = jest.fn();
    TestModal = ({ onAgree, onDisagree, visible }) => {
      return (
        <div class="modal-open">
          <Modal moreClasses={["test-modal"]} visible={visible}>
            <Header class="modal-header">
              <h3 class="modal-title">This is the header</h3>
            </Header>
            <Body>
              <p>Test content</p>
            </Body>
            <Footer class="modal-footer">
              <button class="btn btn-success" onClick={onAgree}>
                Agree
              </button>
              <button class="btn btn-danger" onClick={onDisagree}>
                Do Not Agree
              </button>
            </Footer>
            <Backdrop moreClasses={["extra-backdrop-class"]} visible={true} zIndex={999} />
          </Modal>
        </div>
      );
    };
  });
  afterEach(() => {
    querySelector(document, "body").removeChild(root);
  });
  describe("with modal not visible", () => {
    let el: ?HTMLElement;

    beforeEach(() => {
      render(<TestModal onAgree={onAgree} onDisagree={onDisagree} visible={false} />, root);
      el = root.querySelector(".modal");
    });
    it("should not exist on the element", () => {
      expect(el).toBeNull();
    });
  });
  describe("with modal visible", () => {
    beforeEach(() => {
      render(<TestModal onAgree={onAgree} onDisagree={onDisagree} visible={true} />, root);
      el = querySelector(root, ".modal");
    });

    it("should render", () => {
      expect(el).toMatchSnapshot();
    });
  });
});
