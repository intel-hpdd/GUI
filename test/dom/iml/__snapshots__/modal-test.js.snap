// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal test with modal visible should render 1`] = `
<div
  class="modal fade in test-modal"
  role="dialog"
  style="display: block;"
  tabindex="-1"
>
  <div
    class="modal-dialog"
    role="document"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <h3
          class="modal-title"
        >
          This is the header
        </h3>
      </div>
      <div
        class="modal-body "
      >
        <p>
          Test content
        </p>
      </div>
      <div
        class="modal-footer"
      >
        <button
          class="btn btn-success"
        >
          Agree
        </button>
        <button
          class="btn btn-danger"
        >
          Do Not Agree
        </button>
      </div>
      <div
        class="modal-backdrop fade in extra-backdrop-class"
        style="z-index: 999;"
      />
    </div>
  </div>
</div>
`;
